pyproject.toml

```diff
@@ -25,32 +26,32 @@ classifiers = [
     "Operating System :: Unix",
     "Operating System :: MacOS",
 ]
-requires-python = ">=3.9"
-dependencies = [
-    "aws-cdk-lib==2.48.0",
-    "aws-cdk.aws-lambda-python-alpha",
-    "constructs>=10.0.0,<11.0.0",
-    "boto3",
-    "opensearch-py",
-    "requests",
-    "python-jose"
-]
 
-[project.optional-dependencies]
-dev = [
-    "pytest==6.2.5",
-    "moto"
-]
+[tool.poetry.dependencies]
+python = ">=3.9,<4"
+# Required for aws-cdk.aws-lambda-python-alpha
+aws-cdk-lib = ">=2.66.0,<3.0.0"
+#aws-cdk-lib = "2.48.0"
+boto3 = "^1.26.78"
```

Comment:
Some of these libraries didn't have versions in the requirements.txt file - this is what poetry defaults to, but if the team would prefer to have these always be the most recent version, I can change it.  This just prevents major version bumps without human intervention.

Reply:
I don't think we actually use boto3 in the package?

Reply:
It shows up in the lambda code, and the tests. So, I could move it out of the main dependencies and into the lambda and test specific dependencies, if that's preferable.

Reply:
Ahh, I did not realize that! Yeah, I would put it into the test dependencies section then.

I like keeping things organized where the imports are actually needed/used if possible. The "base package" should really be only what is required to `poetry install --without-tests && cdk deploy` to get the full stack.

Then I think we have:
* lambda_dev: code for the lambdas, which actually builds and installs itself separate, so we don't need to specify those dependencies in our main app
* tests: everything to run `pytest`
* dev / all: everything to be able to develop locally (everything but lambda_dev I think?)

Reply:
Ok, that sounds reasonable to me! I'll double check the dependencies to see what is actually needed for lambdas vs the cdk deploy, it will be good to have those separate.